internal/controllers/clustertemplate_controller.go

```diff
@@ -327,6 +334,54 @@ func generateTemplateID(ctx context.Context, c client.Client, object *provisioni
 	return nil
 }
 
+// validateTemplateParameterSchema return true if the schema contained in the templateParameterSchema
+// field contains the required mandatory parameters
+// - nodeClusterName
+// - oCloudSiteId
+// - policyTemplateParameters
+// - clusterInstanceParameters
+func validateTemplateParameterSchema(object *provisioningv1alpha1.ClusterTemplate) error {
+	mandatoryParams := []string{"nodeClusterName",
+		"oCloudSiteId",
+		"policyTemplateParameters",
+		"clusterInstanceParameters"}
+	if object.Spec.TemplateParameterSchema.Size() == 0 {
+		return utils.NewInputError("templateParameterSchema is present but empty:")
+	}
+	var missingParameter []string
+	for _, param := range mandatoryParams {
+		_, err := utils.ExtractSubSchema(object.Spec.TemplateParameterSchema.Raw, param)
+		if err != nil && strings.HasPrefix(err.Error(), fmt.Sprintf("subSchema %s does not exist:", param)) {
```

Comment:
Maybe refactor these lines to smth like:
```
if err != nil {
    if strings.HasPrefix(err.Error(), fmt.Sprintf("subSchema %s does not exist:", param)) {
        missingParameter = append(missingParameter, param)
    } else {
        return fmt.Errorf("error extracting subschema at key %s: %w", param, err)
    }
}
```

Reply:
Fixed